view.go

```diff
@@ -166,14 +198,39 @@ func (v *View) stop() {
 			}(i, par)
 		}
 		wg.Wait()
+
+		if !v.opts.restartable {
+			v.close()
+		}
 		v.opts.log.Printf("View: shutdown complete")
 	})
 }
 
-// Stop stops the view, frees any resources + connections to kafka
+// close closes all storage partitions
+func (v *View) close() {
+	for _, p := range v.partitions {
+		v.errors.Collect(p.st.Close())
+	}
+}
+
+// Stop stops the view, closes storage partitions, and frees resources
 func (v *View) Stop() {
 	v.opts.log.Printf("View: stopping")
+
+	// do not allow any reinitialization
+	v.mInit.Lock()
+	if v.terminated {
+		return
+		v.mInit.Unlock()
```

Comment:
This unlock is dead code and could result in a deadlock.

Reply:
damn it. Thanks